File: Data Structures & Algorithms/anagram-groups/submission-1.java
class Solution {
    public List<List<String>> groupAnagrams(String[] strs) {

        Map <String, List<String> > myMap = new HashMap<>();
        List < List<String> > result = new ArrayList<>();

        for (String item : strs){

            char [] myarray = item.toCharArray();
            Arrays.sort(myarray);
            String temp = Arrays.toString(myarray);

            if (!myMap.containsKey(temp)){
                myMap.put(temp, new ArrayList<String>());
            }

            myMap.get(temp).add(item);
        }

        for (String i : myMap.keySet()){
            result.add(myMap.get(i));
        }

        return result;

    }
}

File: Data Structures & Algorithms/products-of-array-discluding-self/submission-1.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        
        result = []
        p = 1

        for i in range (len(nums)):
            
            for j in range (len(nums)):
                if i == j:
                    continue
                
                p *= nums[j]
            
            result.append(p)
            p=1

        return result